fission/src/mirabuf/ProtectedZoneSceneObject.ts

```diff
@@ -115,40 +138,9 @@ class ProtectedZoneSceneObject extends SceneObject {
                         this.zoneCollision(body1)
                     }
 
-                    // If the preference is set to require robot contact, we want to penalize robots here
-                    if (!this._prefs?.requireRobotContact) return
-                    const [collisionObjectBody1, collisionObjectBody2] = [body1, body2].map(body => {
-                        const associate = World.physicsSystem.getBodyAssociation(body) as RigidNodeAssociate | undefined
-                        return associate?.sceneObject as MirabufSceneObject | undefined
-                    })
-                    if (!collisionObjectBody1 || !collisionObjectBody2) return
-                    // Makes sure that both robots are from opposing alliances
-                    if (collisionObjectBody1.alliance === collisionObjectBody2.alliance) return
-                    // Ensure that both bodies are robots are inside the zone
-                    if (
-                        Date.now() - (this._robotsInside.get(collisionObjectBody1) ?? 0) > 500 ||
-                        Date.now() - (this._robotsInside.get(collisionObjectBody2) ?? 0) > 500
-                    ) {
-                        return
-                    }
-                    // Ensures that infinite collisions do not occur
-                    if (Date.now() - this._lastRobotCollisionTime < 1000) return
-                    this._lastRobotCollisionTime = Date.now()
-
-                    // Penalize the robot that entered the opposing alliance protected zone
-                    if (collisionObjectBody1.alliance === this._prefs?.alliance) {
-                        SimulationSystem.robotPenalty(
-                            collisionObjectBody2,
-                            this._prefs?.penaltyPoints ?? 0,
-                            `Touched robot in protected zone`
-                        )
-                    } else {
-                        SimulationSystem.robotPenalty(
-                            collisionObjectBody1,
-                            this._prefs?.penaltyPoints ?? 0,
-                            `Touched robot in protected zone`
-                        )
-                    }
+                    // Handle contact-based penalties based on the configured contact type
+                    if (this._prefs?.contactType == ContactType.ROBOT_ENTERS || !this.isZoneActive()) return
+                    this.handleContactPenalty(body1, body2)
```

Comment:
This doesn't seem to trigger if the contact starts legally (neither robot in the zone) but then, without breaking contact, one robot pushes the other/both of them into the zone such that they are now in violation of the zone's rules.